### PR TITLE
Other Schools / Courses component

### PR DESCRIPTION
**Pull request by jleong01**
Initial implementation of other school courses tiles component

### CODE REVIEW

File: core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Optional
+    @Inject
+    @Via("resource")
+    private List<Resource> schoolPaths;
+
+    @Inject
+    private Page currentPage;
+
+    private static final int MAX_TILES = 3;
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java]: move this constant to Constants.java

File: core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (!this.tileListingItems.isEmpty()) {
+            return this.tileListingItems;
+        }
+
+
+        if (null != schoolPaths && !schoolPaths.isEmpty()) {
+
+            for (Resource tileResource : schoolPaths) {
+                LOGGER.debug("Tile items are :", tileResource);
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java]: LOGGER.debug("Tile items are : {}", tileResource);

File: core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                TileListingModelBean theTile = tileResource.adaptTo(TileListingModelBean.class);
+
+
+                Page contentPage = tileResource.getResourceResolver()
+                                                .resolve(theTile.getTitleLink())
+                                                .adaptTo(Page.class);
+                theTile.setTitle(null != contentPage.getTitle() ?  contentPage.getTitle()
+                		: contentPage.getPageTitle());
+                theTile.setDescription(contentPage.getDescription());
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java]: don't use setter methods in Bean. Use the parameterized constructor instead.

File: core/src/main/java/com/taylorsuniversity/models/bean/TileListingModelBean.java [hunk 4]
```diff
@@ -8,6 +8,8 @@
 import org.apache.sling.models.annotations.Model;
 import org.apache.sling.models.annotations.Optional;
 import org.apache.sling.api.resource.ResourceResolver;
+
```
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/TileListingModelBean.java]: Remove the setter methods from this class.

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/content/othercoursestiles/_cq_editConfig.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    cq:dialogMode="floating"
+    cq:layout="editbar"
+    jcr:primaryType="cq:EditConfig">
+    <cq:listeners
+        jcr:primaryType="cq:EditListenersConfig"
+        afteredit="REFRESH_PAGE"/>
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/content/othercoursestiles/_cq_editConfig.xml]: REFRESH_SELF

File: core/src/main/java/com/taylorsuniversity/models/bean/TileListingModelBean.java (hunk 4 above)
Comment (bsinghal1) [inline core/src/main/java/com/taylorsuniversity/models/bean/TileListingModelBean.java]: Remove the "Resource.class" annotation from this class.

Comment (jleong01) [reply core/src/main/java/com/taylorsuniversity/models/bean/TileListingModelBean.java]: updated nad removed to newly created bean OtherCourseListingBean

Comment (jleong01) [reply core/src/main/java/com/taylorsuniversity/models/bean/TileListingModelBean.java]: setters removed on newly created bean OtherCourseListingBean

File: core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java (hunk 3 above)
Comment (jleong01) [reply core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java]: updated constructor on newly created bean OtherCourseListingBean

File: core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java (hunk 2 above)
Comment (jleong01) [reply core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java]: logger changed.

File: core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java (hunk 1 above)
Comment (jleong01) [reply core/src/main/java/com/taylorsuniversity/models/OtherCourseModel.java]: moved to constants.java

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/content/othercoursestiles/_cq_editConfig.xml (hunk 5 above)
Comment (jleong01) [reply ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/content/othercoursestiles/_cq_editConfig.xml]: updated to REFRESH_SELF

**Pull request merged**